Rework planning for foreman

maximiliankolb opened the pull request:
#### What changes are you introducing?

* Rework "Planning for Foreman" guide to make it fit for all flavours
* Add images for upstream+downstream orcharhino

#### Why are you introducing these changes? (Explanation, links to references, issues, etc.)

* Show it upstream for all flavours
* Make content fit for downstream orcharhino

#### Anything else to add? (Considerations, potential downsides, alternative solutions you have explored, etc.)

* Requires https://github.com/theforeman/foreman-documentation/pull/3632
* Requires https://github.com/theforeman/foreman-documentation/pull/3639
* Style-wise, I will try to apply all your feedback.
* **The guide is big: I might create issues for existing technical inaccuracies rather than fixing/retesting everything myself.**
* Created images for upstream for all but two graphics that contain too much information IMHO:
  * `guides/common/images/topology-direct-satellite.png`
  * `guides/common/images/topology-isolated-satellite.png`
* I have split two files that had different headings to snippets and small concept files with correct headings and anchors (diff on GitHub might look weird):
  * `guides/common/modules/con_creating-and-managing-roles.adoc` -> `guides/common/modules/snip_creating-and-managing-roles.adoc` which is now included in two `con_` files
  * `guides/common/modules/ref_overview-of-authentication-methods-in-project.adoc` -> `guides/common/modules/snip_table-authentication-methods.adoc` which is also included in two `con_` files

#### Checklists

* [x] I am okay with my commits getting squashed when you merge this PR.
* [x] I am familiar with the [contributing](https://github.com/theforeman/foreman-documentation/blob/master/CONTRIBUTING.md) guidelines.

no cherry-picks

Lennonka commented on guides/common/assembly_provisioning-requirements.adoc:
```diff
@@ -0,0 +1,17 @@
+
+include::modules/con_http-booting-requirements-with-unmanaged-dhcp.adoc[leveloffset=+2]
+
+ifdef::foreman-el,katello,orcharhino[]
```
That is correct. However, I need to take a look at the content of that module. I'll do that in a separate PR.

Lennonka commented on guides/doc-Planning_for_Project/master.adoc:
```diff
@@ -5,18 +5,13 @@ include::common/header.adoc[]
 
 include::common/modules/con_foreman-overview-and-concepts.adoc[]
 
+ifdef::katello,satellite,orcharhino[]
```
Missing `endif` below Content and patch management?

There's not much in the foreman-el build right now :)

maximiliankolb replied:
Yes, my bad. Moved it.

Lennonka commented on guides/common/images/lifecycle.svg:
There's a leftover something on the left.

Lennonka commented on guides/common/images/lifecycle.svg:
"Foreman Server & Smart Proxy Servers" -- Foreman Server is not involved, it's only "Smart Proxy servers".
(You might have thought that because Satellite sometimes used to call Capsules "Red Hat Satellite Capsules", but that doesn't mean there's Satellite Server, it's just a silly marketing name leftover.)

Lennonka commented on guides/common/images/system-architecture.svg:
"Foreman Server & Smart Proxy Servers" -- Foreman Server is not involved, it's only "Smart Proxy servers".

Lennonka commented on guides/common/modules/con_content-types-in-project.adoc:
```diff
@@ -6,7 +6,7 @@ ifdef::satellite[]
 endif::[]
 ifdef::katello,orcharhino[]
-You can use content from Red{nbsp}Hat as well as from Canonical, Oracle, SUSE, and other custom content.
+You can use any custom content and content from Red{nbsp}Hat, Canonical, Oracle, and SUSE.
```
Adding this is okay. I'm not sure if it's necessary to mention other content specifically.

Lennonka reviewed (commented):
That's a hell of work, Max!
Some initial thoughts below.

The graphics could use a bit of designer's touch, but I can tackle that later in a follow-up PR.

aneta-petrova commented on guides/common/modules/con_content-view-scenarios.adoc:
```diff
@@ -8,40 +8,63 @@ It is not recommended to associate hosts directly with the Library as it prevent
-* *A single lifecycle environment path* – both operating system and applications content is promoted through the same path.
-The path can consist of several stages (for example *Development*, *QA*, *Production*), which enables thorough testing but requires additional effort.
+* *A single lifecycle environment path* {endash} both operating system and applications content is promoted through the same path.
+The path can consist of several stages, for example, *Development*, *QA*, *Production*, which enables thorough testing but requires additional effort.
```
I think the parentheses were much easier to read. If you want to avoid parentheses at all costs, how about two sentences?

```suggestion
The path can consist of several stages, which enables thorough testing but requires additional effort.
For example, you can use  the *Development*, *QA*, and *Production* stages.
```

aneta-petrova commented on guides/common/modules/con_defining-your-content-library.adoc:
```diff
@@ -32,8 +40,8 @@ You can synchronize repositories manually, or you can create a sync plan to ensu
-endif::[]
-ifndef::katello[]
-* For more information, including procedures for enabling and synchronizing repositories, see {ContentManagementDocURL}Importing_Custom_Content[Importing custom content] in _{ContentManagementDocTitle}_.
+* For more information about basic workflows, see {ContentManagementDocURL}Basic_Content_Management_Workflow_content-management[Basic content management workflow].
+* For more information about Deb content, see {ContentManagementDocURL}Adding_Custom_Deb_Repositories_content-management[Adding Deb repositories] in _{ContentManagementDocTitle}_.
+* For more information about Yum content, see {ContentManagementDocURL}Adding_Custom_RPM_Repositories_content-management[Adding RPM repositories] in _{ContentManagementDocTitle}_.
```
I prefer the original (shorter) list of references because I don't think that a link whose label says "Basic workflow" needs to be introduced with the words "basic workflows" :) And it's similar for the other links. But that's just FYI to explain why I wrote the original list of references the way it was before. This is your PR and your call, obviously :)

maximiliankolb replied:
I have removed "basic workflow" from the description.

aneta-petrova commented on guides/common/modules/con_host-group-structures.adoc:
```diff
@@ -5,20 +5,28 @@ The fact that host groups can be nested to inherit parameters from each other al
 In this hierarchy, the first host group level is reserved for parameters specific to a lifecycle environment.
 The second level contains operating system related definitions, and the third level contains application specific settings.
-Such structure is useful in scenarios where responsibilities are divided among lifecycle environments (for example, a dedicated owner for the *Development*, *QA*, and *Production* lifecycle stages).
+Such structure is useful in scenarios where responsibilities are divided among lifecycle environments, for example, a dedicated owner for the *Development*, *QA*, and *Production* lifecycle stages.
```
Is it just me or does this also look strange without the parentheses? Written this way, it looks like you're listing examples of lifecycle environments (nonsense, I know, but that's what it looks like when I read it as one sentence without the extra parenthesis separator). Would it make sense to split this into two sentences as well?

maximiliankolb replied:
Reworded. What do you think now?

aneta-petrova commented on guides/common/modules/con_provisioning-management-with-project-additional-resources.adoc:
```diff
@@ -2,3 +2,6 @@
 
 * For information about how to provision hosts with {Project}, see {ProvisioningDocURL}[_{ProvisioningDocTitle}_].
+ifdef::orcharhino[]
+* For more information about compute resource providers, see xref:sources/compute_resources.adoc[Compute Resources].
```
'Resources' with capital R is intentional?

maximiliankolb replied:
Yes. This is a whole other story: orcharhino docs use title case almost everywhere. Original: https://docs.orcharhino.com/or/docs/sources/compute_resources.html

aneta-petrova reviewed (commented):
This is fantastic, Maximilian! Just nitpicks from me. I'm not officially approving yet to acknowledge that Lena has found a couple of things that would be good to fix.

maximiliankolb reviewed (commented):
Rebased to "master" and applied all suggestions to texts by Anet and Lena. Will fix the images later. Docs are ready for re-review.

aneta-petrova reviewed (approved):
Ack, and the foreman-el issue spotted by Lena also looks fixed now.

maximiliankolb reviewed (commented):
Rebased to HEAD of "master" and applied all suggestions by Anet and Lena. Graphics are now exported with "appearance: light" which should make a bit nicer.

maximiliankolb commented:
Rebased to HEAD of "master".

maximiliankolb commented:
Rebased to HEAD of "master" and added svg files for orcharhino in the last two commits. Using `sed` was possible and made this very easy! :partying_face:

Lennonka commented on guides/doc-Planning_for_Project/master.adoc:
Non-blocking: I think the CLI help appendix should go to the Administering guide because during planning, the user shouldn't be required to actually run any commands.

aneta-petrova commented:
Thanks for the heads up! My ack is already in place, it looks like we are now missing `tech ack`.

goarsna commented on guides/common/modules/con_smartproxy-features.adoc:
```diff
@@ -1,20 +1,15 @@
 
-ifdef::satellite,orcharhino[]
+ifdef::katello,satellite,orcharhino[]
 To mirror content from {ProjectServer}, {SmartProxy} provides the following functionalities:
```
```suggestion
To mirror content from {ProjectServer}, {SmartProxyServers} provide the following functionalities:
```

maximiliankolb replied:
Applied.

goarsna reviewed (approved):
The changes look good from a technical POV despite one suggestion.

maximiliankolb reviewed (commented):
Thanks Goars. I applied your suggestion and rebased to HEAD of "master".

maximiliankolb commented:
part of https://github.com/theforeman/foreman-documentation/issues/396
